crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1796,37 +1799,94 @@ impl AuthorityPerEpochStore {
         C: CheckpointServiceNotify,
     >(
         self: &'a Arc<Self>,
-        transactions: &[VerifiedSequencedConsensusTransaction],
-        end_of_publish_transactions: Vec<VerifiedSequencedConsensusTransaction>,
+        transactions: Vec<SequencedConsensusTransaction>,
         consensus_stats: &ExecutionIndicesWithStats,
         checkpoint_service: &Arc<C>,
         object_store: impl ObjectStore,
-    ) -> SuiResult<(
-        Vec<VerifiedExecutableTransaction>,
-        ConsensusCommitBatch<'a, C>,
-    )> {
+        commit_round: Round,
+        commit_timestamp: TimestampMs,
+        skipped_consensus_txns: &IntCounter,
+    ) -> SuiResult<Vec<VerifiedExecutableTransaction>> {
+        let verified_transactions: Vec<_> = transactions
+            .into_iter()
+            .filter_map(|transaction| {
+                self.verify_consensus_transaction(transaction, skipped_consensus_txns)
+            })
+            .collect();
+        let roots: BTreeSet<_> = verified_transactions
+            .iter()
+            .filter_map(|transaction| transaction.0.transaction.executable_transaction_digest())
+            .collect();
+        let (end_of_publish_transactions, mut sequenced_transactions): (Vec<_>, Vec<_>) =
+            verified_transactions
+                .into_iter()
+                .partition(|transaction| transaction.0.is_end_of_publish());
+
+        PostConsensusTxReorder::reorder(
+            &mut sequenced_transactions,
+            self.protocol_config.consensus_transaction_ordering(),
+        );
+
         let mut batch = self.db_batch();
         let (transactions_to_schedule, notifications, lock_and_final_round) = self
             .process_consensus_transactions(
                 &mut batch,
-                transactions,
+                &sequenced_transactions,
                 &end_of_publish_transactions,
                 checkpoint_service,
                 object_store,
             )
             .await?;
         self.record_consensus_commit_stats(&mut batch, consensus_stats)?;
-        Ok((
-            transactions_to_schedule,
-            ConsensusCommitBatch {
-                epoch_store: self.clone(),
-                checkpoint_service: checkpoint_service.clone(),
-                batch,
-                notifications,
-                end_of_publish_transactions,
-                lock_and_final_round,
+
+        // The last block in this function notifies about new checkpoint if needed
+        let final_checkpoint_round = lock_and_final_round.map(|(_, r)| r);
```

Comment:
What is the difference between this and the previous way of getting `final_checkpoint_round`?

Reply:
lock_and_final_round contains a lock besides the round